src/basic/escape.c

```diff
@@ -505,19 +501,18 @@ char* shell_maybe_quote(const char *s, EscapeFlags flags) {
          * escaping too, but that should be OK. */
 
         for (p = s; *p; p++)
-                if (*p <= ' ' ||
-                    *p >= 127 ||
-                    strchr(SHELL_NEED_QUOTES, *p))
+                if (char_is_cc(*p) ||
```

Comment:
this changes the test a bit. previously all non-asciii chars were escaped, they are no longer. which is probably  OK, but needs a commit msg comment at least